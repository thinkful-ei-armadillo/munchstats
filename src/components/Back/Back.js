import React, { Component } from 'react';

export default class Back extends Component{
  render(){
    return (
      <section className='goBack'>
        <span onClick={() => this.props.history.push(`${this.props.path}`)} className='back_button'><i className="fas fa-chevron-left"></i></span>
      </section>
    );
  }
}