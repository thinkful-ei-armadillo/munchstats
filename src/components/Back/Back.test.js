import React from 'react';
import Back from './Back';
import {MemoryRouter} from 'react-router-dom';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Back />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><Back /></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Back/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});