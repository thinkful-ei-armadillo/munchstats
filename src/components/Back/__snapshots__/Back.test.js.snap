// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Back /> renders the UI as expected 1`] = `
<section
  className="goBack"
>
  <span
    className="back_button"
    onClick={[Function]}
  >
    <i
      className="fas fa-chevron-left"
    />
  </span>
</section>
`;
